octobot_trading/personal_data/orders/types/market/market_order.py

```diff
@@ -13,20 +13,19 @@
 #
 #  You should have received a copy of the GNU Lesser General Public
 #  License along with this library.
-import asyncio
-import octobot_commons.asyncio_tools as asyncio_tools
-
 import octobot_trading.enums as enums
 import octobot_trading.personal_data.orders.order as order_class
 
 
 class MarketOrder(order_class.Order):
     async def update_order_status(self, force_refresh=False):
         if self.trader.simulate:
-            asyncio.create_task(self.on_fill(force_fill=True))
-            # In trading simulation wait for the next asyncio loop iteration to ensure this order status
-            # is updated before leaving this method
-            await asyncio_tools.wait_asyncio_next_cycle()
+            # TODO: ensure no issue un not running it in task anymore
+            await self.on_fill(force_fill=True)
+            # asyncio.create_task(self.on_fill(force_fill=True))
```

Comment:
Should we merge it? Are we sure?

Reply:
I didn't spot any issue regarding this, I think we can